From 9697aac8178aec764175f65098ba2bd50267cc5c Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 20 Mar 2024 17:46:06 +0200
Subject: [PATCH] [SW-156330] hl: ppboot: fixed tpm PCR extend prints

tpm_pcr_extend is used to extend PCR for both
ppboot and preboot image. Therefore modifying
the print statement in tpm_pcr_extend to generic one.

Change-Id: I9103d0b538eb8b666f93f2b62ffad9f66808704c
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/lib/common/tpm.c b/lib/common/tpm.c
index 3d64564..aa0e1df 100644
--- a/lib/common/tpm.c
+++ b/lib/common/tpm.c
@@ -63,8 +63,7 @@
 	if (tpm_startup_done == true) {
 		preboot_pcr_idx = get_preboot_pcr_index();
 
-		if (tpm_pcr_extend(preboot_pcr_idx, hash))
-			preboot_puts("PPBOOT: preboot pcr extend failed\n");
+		tpm_pcr_extend(preboot_pcr_idx, hash);
 	}
 }
 
diff --git a/lib/common/tpm_auth.c b/lib/common/tpm_auth.c
index 0adfc82..eec8578 100644
--- a/lib/common/tpm_auth.c
+++ b/lib/common/tpm_auth.c
@@ -106,7 +106,7 @@
 
 	ret = tpm2_pcr_extend(image_index, TPM2_ALG_SHA384, hash, SHA384_SUM_LEN);
 	if (ret) {
-		preboot_puts("preboot_pcr_extend: pcr extend failed\n");
+		preboot_puts_hex("pcr extend failed for image index: ", image_index, false);
 		cpu_boot_err_set(CPU_BOOT_ERR_TPM_FAIL);
 	}
 
